zerver/tests/test_subs.py

```diff
@@ -575,6 +579,7 @@ def test_create_stream_by_admins_only_setting(self):
 
         # Change setting back to default
         do_set_realm_create_stream_by_admins_only(user_profile.realm, False)
+        do_set_realm_waiting_period_threshold(user_profile.realm, initial_threshold)
```

Comment:
Your testing doesn't verify all the possible cases -- it should verify:
* no threshhold (i.e. previous behavior)
* threshhold set, before the waiting period has passed
* threshhold set, after the waiting period has passed